docs/en/guides/pipeline-integrations/development/sgtk-developer-app.md

```diff
@@ -0,0 +1,257 @@
+You also need to think about the context that your app will depend on.
+For example, can your app only run when you know the task the user is working on, or can it run with just the project known?
+Knowing this will dictate which environment YAMLs and engines you need to add your app settings to.
+
+If you're not sure right now, it's a good idea to start by adding it to the `tk-shell` engine in the project environment, that way you can [run it from your IDE](./sgtk-developer-bootstrapping.md) or via the command line with the tank command if you have a [centralized config](https://developer.shotgunsoftware.com/tk-core/initializing.html#centralized-configurations), which will make it quicker to develop with.
+
+To start with you will use a [dev descriptor](https://developer.shotgunsoftware.com/tk-core/descriptor.html#pointing-to-a-path-on-disk) for the location of your app.
+
+```yaml
+tk-multi-starterapp:
+  location:
+    type: dev
+    path: /path/to/source_code/tk-multi-starterapp
+```
+This instructs Toolkit to load the app code directly from disk in the given location, which is great for development, where you want to change the code all the time.
+Later when you add the app to your production config, you may want to use a different descriptor.
+
+Now you've added the app to an environment, you should be able to go ahead and launch it. How you launch it will depend on the engine and environment you defined it in.
+
+## Part 4: Developing the app
+
+You are now all set to start developing your app!
+
+### Anatomy of the Template Starter App
+
+The [template starter app](https://github.com/shotgunsoftware/tk-multi-starterapp) contains the following basic structure:
+
+![](./images/tk-multi-starterapp-directory.png)
+
+- **app.py** - The app entry point and menu registration can be found in the `app.py` file. This is where you typically set up your classes and get things initialized and get menu items registered.
+- **info.yml** - Also known as the manifest file, defines all the different settings that this app requires when it is installed. 
+Settings are often useful if you want reusable apps and you don't want to hard code any values in the app itself.
+- **python/app/dialog.py** - This contains the logic and event callbacks etc. that produce the main app window.
+- **python/app/ui** - This folder contains the automatically generated UI code and resource file. You don't edit this directly, instead, you edit the Qt UI file in the resources folder.
+- **resources/** - In the resources folder, the `dialog.ui` file is a Qt Designer file that you can open up and use to rapidly design and define the look and feel of the app. 
+Once you have made changes you have to execute the `build_resources.sh` script which will convert the UI file to python code and store it as the `/python/app/ui/dialo.py`.
```

Comment:
- comma after "changes", and another after "script" 
 - is there a "g" missing in `/python/app/ui/dialo.py`?

> store it as the `/python/app/ui/dialo.py`
 - I'd either get rid of "the", or say "the `python/app/ui/dialog.py` *file*"